nntrainer/compiler/multiout_realizer.cpp

```diff
@@ -9,16 +9,99 @@
  * @author Jihoon Lee <jhoon.it.lee@samsung.com>
  * @bug No known bugs except for NYI items
  */
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+#include <common_properties.h>
 #include <compiler_fwd.h>
+#include <layer_node.h>
 #include <multiout_realizer.h>
+#include <remap_realizer.h>
 
 namespace nntrainer {
 MultioutRealizer::~MultioutRealizer() {}
 
 GraphRepresentation
 MultioutRealizer::realize(const GraphRepresentation &reference) {
+  GraphRepresentation processed(reference.begin(), reference.end());
+
+  std::unordered_multiset<std::string> freq_map;
+  std::unordered_set<std::string> node_names;
+  std::unordered_set<std::string> connection_names;
+
+  /// 1. build frequency map and connection names
+  for (auto &node : reference) {
+    NNTR_THROW_IF(node_names.count(node->getName()), std::invalid_argument)
+      << "node name clashes: " << node->getName();
+    node_names.emplace(node->getName());
+
+    for (unsigned int i = 0, num_nodes = node->getNumInputConnections();
+         i < num_nodes; ++i) {
+      props::InputConnection c(props::Connection(
+        node->getInputConnectionName(i), node->getInputConnectionIndex(i)));
+      auto uniq_name = to_string(c);
+      connection_names.emplace(uniq_name);
+      freq_map.emplace(uniq_name);
+    }
+  }
+
+  /// 2. for each connection names, if a connection is referenced multiple
+  /// times, create multioutput node and remap to multi output node index
+  std::unordered_multimap<std::string /**< original id */,
+                          std::shared_ptr<LayerNode> /**< created node */>
+    multiout_nodes;
+
+  for (auto &con_name : connection_names) {
+    props::InputConnection con;
+    from_string(con_name, con);
+    if (freq_map.count(con_name) <= 1) {
```

Comment:
`== 1`? < 1 should be error?

Reply:
con_name == 0 means it's input, output or dangled. Multiout realizer cannot tell if it's error or intended.

Reply:
Let's just add a note here that `<1` should not happen and someone must check for this and report error.

Reply:
Done.

```cpp
    /// @note freq < 1 should never happen as the map entry is not created.
    /// but if it happens multiout realizer is not interested in checking if it
    /// is a dangled or actually an output. So there is no assurance done at
    /// this point. Some other class must check if the given graph is formed in
    /// a correct way.
```

Reply:
note that #1734 which was ahead of this PR was merged. This PR will contain only review changes, and won't be referenced at other PR's

Reply:
thanks.